src/inputs/ObjectSearch.tsx

```diff
@@ -288,13 +288,13 @@ class ObjectSearch extends Component<IObjectSearchProps> {
 
   public render () {
     const { id, onAddNew, searchOnEmpty, disabled } = this.injected
-      , isLoading = this.isLoading.isTrue
-      , canSearch = this.hasSearch || searchOnEmpty
-      , showNoResultsOption = canSearch && !isLoading && !this.hasOptions
-      , showAddOption = this.hasSearch && onAddNew
-      , showNoSearch = !this.hasSearch
+      , isLoading: boolean = this.isLoading.isTrue
+      , canSearch: boolean = this.hasSearch || !!searchOnEmpty
+      , showNoResultsOption: boolean = canSearch && !isLoading && !this.hasOptions
+      , showAddOption: boolean = !!(this.hasSearch && onAddNew)
+      , showNoSearch: boolean = !this.hasSearch
```

Comment:
It seems like a lot of these type annotations aren't necessary if these dependent getters and booleans (e.g. `SmartBool`, `this.hasSearch`) are annotated properly

Reply:
They're for catching short-circut-logic problems that humans don't catch well. For example, line 294 originally resulted in a boolean OR a function. Same with 292. These aggressive types aren't to help the compiler, they're to catch assumptions by the programmer.

Reply:
Fair enough!